java/Cherepnin_Oleh/src/com/zoolatech/lecture6/tasks/_3/UserService.java

```diff
@@ -0,0 +1,21 @@
+package com.zoolatech.lecture6.tasks._3;
+
+import com.zoolatech.lecture6.tasks._3.storage.impl.UserCacheImpl;
+import com.zoolatech.lecture6.tasks._3.storage.impl.UserTableImpl;
+
+public class UserService {
+    private UserRepository repository;
+
+    {
+        repository = new UserRepository(new UserCacheImpl(), new UserTableImpl());
+    }
+
+    public String findUserEmail(String id) {
+        try {
+            return repository.findUserEmail(id);
+        } catch (UserMissingException exception) {
+            return exception.getMessage();
```

Comment:
`(it’s enough to print information to the output).`

it's better to not return exception messages as a result of the method call, cause it can be mixed up with the valid method result (emails in a current case).

Reply:
```
"Something went wrong...";
```
is also not a valid method result